File: CS/CreateReportFromRichEdit/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.IO;
using DevExpress.XtraRichEdit;
using DevExpress.XtraReports.UI;
using DevExpress.XtraRichEdit.API.Native;

namespace CreateReportFromRichEdit {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void simpleButton1_Click(object sender, EventArgs e) {
            OpenDocumentAndShowReport();
        }

        void OpenDocumentAndShowReport() {
            OpenFileDialog openFileDialog1 = new OpenFileDialog();

            openFileDialog1.Filter = "txt files (*.txt)|*.txt|All files (*.*)|*.*";
            openFileDialog1.FilterIndex = 2;
            openFileDialog1.RestoreDirectory = true;

            if (openFileDialog1.ShowDialog() == DialogResult.OK) {
                try {
                    if (openFileDialog1.OpenFile() != null) {
                        using (RichEditDocumentServer reds = new RichEditDocumentServer()) {
                            reds.LoadDocument(openFileDialog1.FileName);

                            using (XtraReport1 report = new XtraReport1()) {
                                report.RichText.Rtf = reds.RtfText;
                                report.ShowPreviewDialog();
                            }
                        }
                    }
                } catch (Exception ex) {
                    MessageBox.Show("Error: Could not read file from disk. Original error: " + ex.Message);
                }
            }

        }
    }
}

File: VB/CreateReportFromRichEdit/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports System.IO
Imports DevExpress.XtraRichEdit
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraRichEdit.API.Native

Namespace CreateReportFromRichEdit
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
			OpenDocumentAndShowReport()
		End Sub

		Private Sub OpenDocumentAndShowReport()
			Dim openFileDialog1 As New OpenFileDialog()

			openFileDialog1.Filter = "txt files (*.txt)|*.txt|All files (*.*)|*.*"
			openFileDialog1.FilterIndex = 2
			openFileDialog1.RestoreDirectory = True

			If openFileDialog1.ShowDialog() = System.Windows.Forms.DialogResult.OK Then
				Try
					If openFileDialog1.OpenFile() IsNot Nothing Then
						Using reds As New RichEditDocumentServer()
							reds.LoadDocument(openFileDialog1.FileName)

							Using report As New XtraReport1()
								report.RichText.Rtf = reds.RtfText
								report.ShowPreviewDialog()
							End Using
						End Using
					End If
				Catch ex As Exception
					MessageBox.Show("Error: Could not read file from disk. Original error: " & ex.Message)
				End Try
			End If

		End Sub
	End Class
End Namespace
